libcnb/examples/example-02-ruby-sample/src/layers/bundler.rs

```diff
@@ -56,50 +64,38 @@ impl LayerLifecycle<RubyBuildpack, BundlerLayerMetadata, HashMap<String, String>
             .wait()?;
 
         if !bundle_exit_code.success() {
-            return Err(anyhow::anyhow!("Could not install gems!"));
+            return Err(anyhow::anyhow!("Could not bundle install"));
         }
 
-        Ok(LayerContentMetadata {
-            types: LayerTypes {
-                build: false,
-                launch: true,
-                cache: true,
-            },
-            metadata: BundlerLayerMetadata {
-                gemfile_lock_checksum: sha256_checksum(build_context.app_dir.join("Gemfile.lock"))?,
-            },
+        LayerResultBuilder::new(BundlerLayerMetadata {
+            gemfile_lock_checksum: sha256_checksum(context.app_dir.join("Gemfile.lock"))?,
         })
+        .build()
     }
 
-    fn validate(
+    fn existing_layer_strategy(
         &self,
-        _layer_path: &Path,
-        layer_content_metadata: &LayerContentMetadata<BundlerLayerMetadata>,
-        build_context: &BuildContext<RubyBuildpack>,
-    ) -> ValidateResult {
-        let checksum_matches = sha256_checksum(build_context.app_dir.join("Gemfile.lock"))
-            .map(|local_checksum| {
-                local_checksum == layer_content_metadata.metadata.gemfile_lock_checksum
-            })
-            .unwrap_or(false);
-
-        if checksum_matches {
-            ValidateResult::KeepLayer
-        } else {
-            ValidateResult::UpdateLayer
-        }
+        context: &BuildContext<Self::Buildpack>,
+        layer: &LayerData<Self::Metadata>,
+    ) -> anyhow::Result<ExistingLayerStrategy> {
+        sha256_checksum(context.app_dir.join("Gemfile.lock")).map(|checksum| {
```

Comment:
Previously the implementation here would `Update` if there was an error in `sha256_checksum` reading `Gemfile.lock`, now it errors, which seems preferred. However I imagine the error message will be hard to debug due to the anyhow usage. All the more reason for us to remove anyhow from this example in the future :-) 

(no action needed here now)

Reply:
Yeah, we spoke about this before, I filed #167 for it.